### PR TITLE
Add a new mode, `json`, which supports SPI image labeling.  It's an alternative to `label` mode, but outputs labels into a json file.

### PR DESCRIPTION
**Pull request by carbonscott**
Supported inputs for SPI image labeling after psocake GUI is open:
- Key 0: No hit.
- Key 1: Single hit.
- Key 2: Multiple hits.
- Key 3: Unknown hits.
- Key 4: No hit (alias to Key 0, for keeping user's hand at key 1-4 area)
- Key s: Save a json file with the extension `.label.json`.

Json file format:
- Filename follows the convention: `{exp}.{run}.label.json`, such as
  `amo06516.0102.label.json`.
- If `{exp}.{run}.label.json` has existed, a backup file with a
  timestamp will be created automatically when `s` is pressed.
  Backup file has a filename following convention:
  `{exp}.{run}.{timestamp}.label.json.bak`, such as
  `amo06516.0102.20220113171435.label.json.bak`.
- File content consists of key value pairs -- `{eventNumber}:{label}`.

### CODE REVIEW

File: psocake/gui.py [hunk 1]
```diff
@@ -491,6 +567,20 @@ def mouseClicked(evt):
 
 def main():
     global ex
+    global eventDocument
+    eventDocument = {}
+    global hit_dict
+    hit_dict = { QtCore.Qt.Key_4 : 0,   # No     hit (alias)
```
Comment (chuckie82) [inline psocake/gui.py:573]: No hit (alias) is confusing. Why is this needed?

File: psocake/gui.py [hunk 2]
```diff
@@ -491,6 +567,20 @@ def mouseClicked(evt):
...
+                 QtCore.Qt.Key_0 : 0,   # No     hit
+                 QtCore.Qt.Key_1 : 1,   # Single hit
+                 QtCore.Qt.Key_2 : 2,   # Multi  hit
+                 QtCore.Qt.Key_3 : 3, } # Unknow hit
+    global hit_interpret_dict
+    hit_interpret_dict = { 0 : 'NO',
```
Comment (chuckie82) [inline psocake/gui.py:579]: Hard coding these labels is ok for SPI labeling. But what if another psocake user wants to label something else for machine learning? A generalizable interface would be nice.

File: psocake/gui.py (hunk 1 above)
Comment (carbonscott) [reply psocake/gui.py:573]: Key 0 is on the far right of the keyboard.  The alias allows a user to keep his/her fingers at Key 1-4 area.   This feature is not necessary for Numpad users.

File: psocake/gui.py (hunk 2 above)
Comment (carbonscott) [reply psocake/gui.py:579]: That's a good point I missed.  Hmm... there needs to be a better solution.  Maybe like a prompt for user to type?

**Comment by carbonscott** [comment]
Image labeling is not limited to SPI scenario only, but suitable for general purpose.  Two user prompts are included in this version.  

- `l` for enabling a GUI dialog to input a label.
- `g` for enabling a GUI dialog to input a specific event that a user wants to go to.

**Comment by carbonscott** [comment]
This feature may have been project-specific, and is no longer needed.
